Integration testing and minor readme updates

agitter opened the pull request:
This pull request adds a skeleton integration testing framework for Travis CI.  The only test currently is `test_import.py`, which confirms that the dependencies were installed correctly.  It is possible to add new test functions that would either be unit tests for specific functions or integration tests that confirm the results generated on the Tox21 data, for example, are similar to the stored results.

There are also minor updates to the readme.

Before merging, you can enable Travis CI testing [with these instructions](https://docs.travis-ci.com/user/getting-started/).

There are a few other small issues to address before merging:
- Does lightchem support both Python 2 and 3?  We can either test both or add the Python 2 requirement to the readme.
- Adding rdkit as a requirement in `setup.py` [caused problems with my Travis CI build](https://travis-ci.org/agitter/lightchem/jobs/186064776).  We can ignore this or try to debug it further.  There is also a newer version of rdkit in the [rdkit channel](https://anaconda.org/rdkit/rdkit).
- Add the Travis CI build status icon once testing is enabled in the main repository.

agitter commented:
I updated the test scripts to include `test_muv.py`.  This is a partially implemented test that runs the MUV example.  It does not yet compare the output files that are generated with the stored files.  It should write the output to a temp directory and then delete the temp directory.

agitter commented:
The MUV test now only tests a single target.  The Travis CI builds are timing out due to a lack of output so we may need to add some intermediate output in `muv_xgboost_models.py` to keep them running.  `test_muv.py` runs one of of my local machines in about 7 minutes.

haozhenWu commented:
@agitter I think we can specify to use python2 for now. 
Rdkit is only used to generate fingerprint. We can ignore it now and I will debug later.

haozhenWu commented:
@agitter I successfully tested the 'pytest' of your branch locally. I am wondering how can I test the TRAVIS CI ? Besides these, everything looks good. Should I merge your branch first or we can continue commit to your branch? Like changing the PCBA and TOX21 version ?

agitter commented:
@haozhenWu I don't remember why I changed the `cd` in the example instructions.  I reverted that change in ce6cfec46414c06ad81383410f837735438ac1d2  Does it look okay now?

If everything looks good, I think you can merge this branch now and then work on any other pending changes later when time permits.  This pull request has many commits already, and I find it easier to review and merge smaller batches of changes.

To test Travis CI, you'll have to link your GitHub account to Travis CI and give it permission to watch your commits.  The guide here gives some specific steps https://docs.travis-ci.com/user/getting-started/
- Sign in to Travis CI with your GitHub account and accept the permissions
- From your Travis CI profile page, enable testing on the lightchem repository

That should be all you need because I already have the `.travis.yml` file as part of this pull request.  Then every time you push to the lightchem repository or every time someone makes a pull request, Travis CI will execute the tests that are run with `pytest`.
